cmd/solana_exporter/exporter_test.go

```diff
@@ -0,0 +1,163 @@
+# TYPE solana_active_validators gauge
+solana_active_validators{state="current"} 2
+solana_active_validators{state="delinquent"} 1
+`,
+		"solana_validator_activated_stake": `
+# HELP solana_validator_activated_stake Activated stake per validator
+# TYPE solana_validator_activated_stake gauge
+solana_validator_activated_stake{nodekey="4MUdt8D2CadJKeJ8Fv2sz4jXU9xv4t2aBPpTf6TN8bae",pubkey="xKUz6fZ79SXnjGYaYhhYTYQBoRUBoCyuDMkBa1tL3zU"} 49
+solana_validator_activated_stake{nodekey="B97CCUW3AEZFGy6uUg6zUdnNYvnVq5VG8PUtb2HayTDD",pubkey="3ZT31jkAGhUaw8jsy4bTknwBMP8i4Eueh52By4zXcsVw"} 42
+solana_validator_activated_stake{nodekey="C97CCUW3AEZFGy6uUg6zUdnNYvnVq5VG8PUtb2HayTDD",pubkey="4ZT31jkAGhUaw8jsy4bTknwBMP8i4Eueh52By4zXcsVw"} 43
+`,
+		"solana_validator_last_vote": `
+# HELP solana_validator_last_vote Last voted slot per validator
+# TYPE solana_validator_last_vote gauge
+solana_validator_last_vote{nodekey="4MUdt8D2CadJKeJ8Fv2sz4jXU9xv4t2aBPpTf6TN8bae",pubkey="xKUz6fZ79SXnjGYaYhhYTYQBoRUBoCyuDMkBa1tL3zU"} 92
+solana_validator_last_vote{nodekey="B97CCUW3AEZFGy6uUg6zUdnNYvnVq5VG8PUtb2HayTDD",pubkey="3ZT31jkAGhUaw8jsy4bTknwBMP8i4Eueh52By4zXcsVw"} 147
+solana_validator_last_vote{nodekey="C97CCUW3AEZFGy6uUg6zUdnNYvnVq5VG8PUtb2HayTDD",pubkey="4ZT31jkAGhUaw8jsy4bTknwBMP8i4Eueh52By4zXcsVw"} 148
+`,
+		"solana_validator_root_slot": `
+# HELP solana_validator_root_slot Root slot per validator
+# TYPE solana_validator_root_slot gauge
+solana_validator_root_slot{nodekey="4MUdt8D2CadJKeJ8Fv2sz4jXU9xv4t2aBPpTf6TN8bae",pubkey="xKUz6fZ79SXnjGYaYhhYTYQBoRUBoCyuDMkBa1tL3zU"} 3
+solana_validator_root_slot{nodekey="B97CCUW3AEZFGy6uUg6zUdnNYvnVq5VG8PUtb2HayTDD",pubkey="3ZT31jkAGhUaw8jsy4bTknwBMP8i4Eueh52By4zXcsVw"} 18
+solana_validator_root_slot{nodekey="C97CCUW3AEZFGy6uUg6zUdnNYvnVq5VG8PUtb2HayTDD",pubkey="4ZT31jkAGhUaw8jsy4bTknwBMP8i4Eueh52By4zXcsVw"} 19
+`,
+		"solana_validator_delinquent": `
+# HELP solana_validator_delinquent Whether a validator is delinquent
+# TYPE solana_validator_delinquent gauge
+solana_validator_delinquent{nodekey="4MUdt8D2CadJKeJ8Fv2sz4jXU9xv4t2aBPpTf6TN8bae",pubkey="xKUz6fZ79SXnjGYaYhhYTYQBoRUBoCyuDMkBa1tL3zU"} 1
+solana_validator_delinquent{nodekey="B97CCUW3AEZFGy6uUg6zUdnNYvnVq5VG8PUtb2HayTDD",pubkey="3ZT31jkAGhUaw8jsy4bTknwBMP8i4Eueh52By4zXcsVw"} 0
+solana_validator_delinquent{nodekey="C97CCUW3AEZFGy6uUg6zUdnNYvnVq5VG8PUtb2HayTDD",pubkey="4ZT31jkAGhUaw8jsy4bTknwBMP8i4Eueh52By4zXcsVw"} 0
+`,
+		"solana_node_version": `
+# HELP solana_node_version Node version of solana
+# TYPE solana_node_version gauge
+solana_node_version{version="1.16.7"} 1
+`,
+	}
+
+	for testName, expectedValue := range testCases {
+		t.Run(
+			testName,
+			func(t *testing.T) {
+				if err := testutil.CollectAndCompare(
+					staticCollector,
+					bytes.NewBufferString(expectedValue),
+					testName,
+				); err != nil {
+					t.Errorf("unexpected collecting result for %s: \n%s", testName, err)
+				}
+			},
+		)
```

Comment:
```suggestion
		t.Run(testName, func(t *testing.T) {
			err := testutil.CollectAndCompare(staticCollector, bytes.NewBufferString(expectedValue), testName)
			assert.NotNilF(t, err, "unexpected collecting result for %s: %s", testName, err)
		})
```

Please don't excessively indent stuff. It makes it much harder to read and is totally unnecessary. This does the same thing, is easier to read, and is withing go formatting standards. If your editor is doing this, please update it to not.

Also, use testify not the testing framework where you can. Remember my saying on having less "code" and more "data" in tests? Using testify, you don't need to have that conditional, so there is less code and more data / assertions.

Also 4 lines < 12 lines of code!